distribution/lib/Standard/Base/0.0.0-dev/src/System/File/File_Permissions.enso

```diff
@@ -0,0 +1,103 @@
+    type Read
+
+    ## Permission for write access for a given entity.
+    type Write
+
+    ## Permission for execute access for a given entity.
+    type Execute
+
+type File_Permissions
+    ## Access permissions for a file.
+    type File_Permissions (owner : Vector Permission) (group : Vector Permission) (others : Vector Permission)
+
+    ## Converts the Enso atom to its Java enum counterpart.
+    to_java : Vector PosixFilePermission
+    to_java =
+        result = Vector.new_builder
+        if self.owner.contains Read then
+            result.append PosixFilePermission.OWNER_READ
+        if self.owner.contains Write then
+            result.append PosixFilePermission.OWNER_WRITE
+        if self.owner.contains Execute then
+            result.append PosixFilePermission.OWNER_EXECUTE
+        if self.group.contains Read then
+            result.append PosixFilePermission.GROUP_READ
+        if self.group.contains Write then
+            result.append PosixFilePermission.GROUP_WRITE
+        if self.group.contains Execute then
+            result.append PosixFilePermission.GROUP_EXECUTE
+        if self.others.contains Read then
+            result.append PosixFilePermission.OTHERS_READ
+        if self.others.contains Write then
+            result.append PosixFilePermission.OTHERS_WRITE
+        if self.others.contains Execute then
+            result.append PosixFilePermission.OTHERS_EXECUTE
+        result.to_vector
+
+    ## Checks if the given file can be read by the owner.
+    owner_read : Boolean
+    owner_read = self.owner.contains Read
+
+    ## Checks if the given file can be written by the owner.
+    owner_write : Boolean
+    owner_write = self.owner.contains Write
+
+    ## Checks if the given file can be executed by the owner.
+    owner_execute : Boolean
+    owner_execute = self.owner.contains Execute
+
+    ## Checks if the given file can be read by the group.
+    group_read : Boolean
+    group_read = self.group.contains Read
+
+    ## Checks if the given file can be written by the group.
+    group_write : Boolean
+    group_write = self.group.contains Write
+
+    ## Checks if the given file can be executed by the group.
+    group_execute : Boolean
+    group_execute = self.group.contains Execute
+
+    ## Checks if the given file can be read by others.
+    others_read : Boolean
+    others_read = self.others.contains Read
+
+    ## Checks if the given file can be written by others.
+    others_write : Boolean
+    others_write = self.others.contains Write
+
+    ## Checks if the given file can be executed by others.
+    others_execute : Boolean
+    others_execute = self.others.contains Execute
+
+## Converts a Java `Set` of Java `PosixFilePermission` to `File_Permissions`.
+from_java_set java_set =
+    owner = Vector.new_builder
+    group = Vector.new_builder
+    others = Vector.new_builder
+
+    if java_set.contains PosixFilePermission.OWNER_READ then
```

Comment:
If the permissions are returned as `Array`, this could have to be changed into a _for cycle_ that iterates the array.

Reply:
I can do it (in fact older variant of the code did that), but why introduce the overhead of converting a Set into an Array if I can pass it directly?

I don't think the code would be any simpler - how would the _for (loop?)_ help? It would still need to check if the given entry is one of the 9 provided and translate according to that.

Reply:
The goal of using an Array was to avoid the Guest value but the elements are all needing to be Guest values so didn't avoid that. 

Will merge as is.

Reply:
The only way to avoid the guest values would probably be to move the whole conversion logic (including instantiating the Enso Atoms representing the options) into the interpreter. But it seems like an overkill.